jupyterdrive/gdrive/drive-contents.js

```diff
@@ -96,438 +90,395 @@ define(function(require) {
-        var contents_prm = metadata_prm.then(function(resource) {
-            that.observe_file_resource(resource);
-            return drive_utils.get_contents(resource, false);
-        });
-
-        return Promise.all([metadata_prm, contents_prm]).then(function(values) {
-            var metadata = values[0];
-            var contents = values[1];
-            var model = files_resource_to_contents_model(path, metadata, contents);
-            return model;
-        });
-    };
-
-
-    /**
-     * Creates a new untitled file or directory in the specified directory path.
+     * For a higher level description on how to use these interfaces, see the
+     * `IContents` interface docs
      *
-     * @method new
-     * @param {String} path: the directory in which to create the new file/directory
-     * @param {Object} options:
-     *      ext: file extension to use
-     *      type: model type to create ('notebook', 'file', or 'directory')
-     */
-    Contents.prototype.new_untitled = function(path, options) {
-        // Construct all data needed to upload file
-        var default_ext = '';
-        var base_name = '';
-        var model = null;
-        if (options['type'] === 'notebook') {
-            default_ext = '.ipynb';
-            base_name = 'Untitled'
-            model = {
-                'type': 'notebook',
-                'content': notebook_model.new_notebook(),
-                'mimetype': drive_utils.NOTEBOOK_MIMETYPE,
-                'format': 'json'
-            };
-        } else if (options['type'] === 'file') {
-            default_ext = '.txt';
-            base_name = 'Untitled';
-            model = {
-                'type': 'file',
-                'content': '',
-                'mimetype': 'text/plain',
-                'format': 'text'
+     **/
+    var GoogleDriveContents = (function () {
+        /**
+         *
+         * A contentmanager handles passing file operations
+         * to the back-end.  This includes checkpointing
+         * with the normal file operations.
+         *
+         * Parameters:
+         *  options: dictionary
+         *      Dictionary of keyword arguments.
+         *          base_url: string
+         **/
+        function GoogleDriveContents(options) {
+            var _this = this;
+            this._base_url = options.base_url;
+            this._config = options.common_config;
+            /**
+             * Stores the revision id from the last save or load.  This is used
+             * when checking if a file has been modified by another user.
+             */
+            this._last_observed_revision = {};
+            var that = this;
+            this._config.loaded.then(function (data) {
+                gapiutils.config(_this._config);
+                gapiutils.gapi_ready.then(driveutils.set_user_info);
+            });
+        }
+        /**
+         * Utility functions
+         */
+        /**
```

Comment:
You have a break in the comment here, not sure how that will render in the end docs.